six/modules/python/six.sicd/source/six_sicd.i

```diff
@@ -39,20 +39,83 @@ using namespace six;
 six::sicd::ComplexData * asComplexData(six::Data* data);
 
 /* Need this because we can't really do it on the python side of things */
-six::sicd::ComplexData * asComplexData(six::Data* data) 
+six::sicd::ComplexData * asComplexData(six::Data* data)
 {
   six::sicd::ComplexData * complexData = dynamic_cast<six::sicd::ComplexData*>(data);
   if( !complexData )
   {
     throw except::BadCastException();
   }
-  else 
+  else
   {
     return complexData;
   }
 }
 
+void writeNITF(std::string fileName, std::vector<std::string> schemaPaths,
+        six::sicd::ComplexData* data, long long imageAdr);
+
+void writeNITF(std::string fileName, std::vector<std::string> schemaPaths,
+        six::sicd::ComplexData* data, long long imageAdr)
+{
+    std::complex<float>* image = reinterpret_cast< std::complex<float>* >(
+            imageAdr);
+
+    six::XMLControlFactory::getInstance().addCreator(
+            six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<six::sicd::ComplexXMLControl>());
+
+    six::Container container(six::DataType::COMPLEX);
+    std::auto_ptr<logging::Logger> logger(logging::setupLogger("out"));
+
+    container.addData(data);
```

Comment:
Take a look at the API but I'm pretty sure the container takes ownership of the data.  I'm assuming that's problematic up a level... you probably need to pass in `data.clone()`.